server/store/event_store.go

```diff
@@ -18,18 +20,29 @@ import (
 const ttlAfterEventEnd = 30 * 24 * time.Hour // 30 days
 const defaultEventTTL = 30 * 24 * time.Hour  // 30 days
 
+type EventMetadata struct {
+	LinkedChannels map[string]struct{}
```

Comment:
Maybe put a doc comment on this field? Conceptually I expected this to be a `[]string` to store the channel ids, but I understand it's a map so it's easier to use for lookup and/or forcing uniqueness. Maybe name it `LinkedChannelIds` to show that there is no other data involved?

Originally when reading the line below, I thought there was a bug where we were accidentally using the index of an array and mistaking that for a channel id, as I thought this was an array before checking the `EventMetadata` struct definition. https://github.com/mattermost/mattermost-plugin-mscalendar/blob/fced1661a8b20065c01e97acacf6944e0ee16486/server/mscalendar/availability.go#L531